src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -166,6 +165,21 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             const auto fullHeight{ ScrollBarCanvas().ActualHeight() };
             const auto totalBufferRows{ update.newMaximum + update.newViewportSize };
 
+            auto drawPip = [&](const auto row, const auto rightAlign, const auto& brush) {
+                Windows::UI::Xaml::Shapes::Rectangle r;
+                r.Fill(brush);
+                r.Width(16.0f / 3.0f); // pip width - 1/3rd of the scrollbar width.
+                r.Height(2);
+                const auto fractionalHeight = row / totalBufferRows;
+                const auto relativePos = fractionalHeight * fullHeight;

```

Comment:
FYI this is where I think we could add our logic to avoid drawing too many of these. By multiplying by the display scale (not sure if that's needed) we can figure out if this pip would overlap with a previously drawn pip. I think the code for that should be somewhat easy... I think? We could use a lambda factory (a lambda that returns a lambda) to do that. Like this:
```cpp
const auto drawPipMaker = [&]() {
    double previousPos = 0;
    return [previousPos, &](const auto row, const auto rightAlign, const auto& brush) mutable {
         if (std::abs(newPos - previousPos) < onePixel)
         {
             return;
         }
         // ...
    };
};
// ...
auto drawPip = drawPipMaker();
// ...
auto drawPip = drawPipMaker();
// ...
```

Reply:
Moved to a follow up in #15057, since the marks aren't _necessarily_ sorted